spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -98,16 +112,44 @@ private void refresh(ConfigStore store, String storeSuffix, String watchedKeyNam
         if (firstTimeMap.get(storeName) == null) {
             storeEtagMap.put(storeName, etag);
             firstTimeMap.put(storeName, false);
-            return;
+            propertyCache.updateRefreshCacheTime(store.getName(), watchedKeyNames, delay);
         }
+        
+        if (!etag.equals(storeEtagMap.get(storeName))) {
+            Date date = new Date();
+
+            // Checks all cached items to see if they have been updated
+            List<String> refreshKeys = propertyCache.getRefreshKeys(store.getName());
+            for (int i = 0; i < refreshKeys.size(); i++) {
+                String refreshKey = refreshKeys.get(i);
+                if (refreshKey.contains(watchedKeyNames.replace("*", ""))) {
+
+                    storeEtagMap.put(storeName, etag);
+                    options = new QueryOptions().withKeyNames(refreshKey)
+                            .withLabels(store.getLabels()).withFields(QueryField.ETAG).withRange(0, 0);
+
+                    keyValueItems = configOperations.getRevisions(store.getName(), options);
+
+                    if (keyValueItems.isEmpty() || keyValueItems.get(0).getEtag()
+                            .equals(propertyCache.getCachedEtag(refreshKey))) {
+                        refreshKeys = propertyCache.updateRefreshCacheTimeForKey(store.getName(), refreshKey, date);
+                        i--;
```

Comment:
The `i--` here sounds special, can we add some comments? Will it make `i` to `-1`?